evalml/tests/integration_tests/test_nullable_types.py

```diff
@@ -92,3 +95,65 @@ def test_imputer_can_impute_features_generated_from_null_email_url_features():
     pl.fit(X, y)
     X_t = pl.transform(X, y)
     assert not X_t.isna().any(axis=None)
+
+
+@pytest.mark.parametrize("problem_type", ProblemTypes.all_problem_types)
+@pytest.mark.parametrize("automl_algorithm", ["iterative", "default"])
+@pytest.mark.parametrize("max_batches", [1, 5, 10])
+@pytest.mark.parametrize("ensembling", [True, False])
+def test_automl_search_with_nullable_types(
+    nullable_type_test_data,
+    nullable_type_target,
+    AutoMLTestEnv,
+    ensembling,
+    max_batches,
+    automl_algorithm,
+    problem_type,
+):
+    if ensembling and is_time_series(problem_type) and automl_algorithm == "default":
+        pytest.skip(
+            "Ensembling is not supported yet for time series in default algorithm.",
+        )
+    X = nullable_type_test_data(has_nans=True)
+
+    if (
+        problem_type == ProblemTypes.MULTICLASS
+        or problem_type == ProblemTypes.TIME_SERIES_MULTICLASS
+    ):
+        y = ww.init_series(pd.Series([0, 1, 2, 0] * 5), logical_type="Categorical")
+    elif (
+        problem_type == ProblemTypes.BINARY
+        or problem_type == ProblemTypes.TIME_SERIES_BINARY
+    ):
+        y = nullable_type_target(ltype="BooleanNullable", has_nans=False)
+    elif (
+        problem_type == ProblemTypes.REGRESSION
+        or problem_type == ProblemTypes.TIME_SERIES_REGRESSION
+    ):
+        y = nullable_type_target(ltype="IntegerNullable", has_nans=False)
+
+    parameters = {}
+    if is_time_series(problem_type):
+        parameters = {
+            "time_index": "dates",
+            "gap": 1,
+            "max_delay": 1,
+            "forecast_horizon": 3,
+        }
+
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type=problem_type,
+        problem_configuration=parameters,
+        optimize_thresholds=False,
+        max_iterations=max_batches,
+        automl_algorithm=automl_algorithm,
+        ensembling=ensembling,
+    )
+    # --> should probably use the automl test env but im not sure if its okay to have the score return value be set?
+    env = AutoMLTestEnv(problem_type)
+    with env.test_context(score_return_value={automl.objective.name: 1.0}):
```

Comment:
I just want to double check that using the automl test env this way doesnt defeat the purpose of this test. I figured we wanted to use it otherwise these tests take forever.